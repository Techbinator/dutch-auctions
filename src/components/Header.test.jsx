import React from "react";
import WrappedHeader, { Header } from "./Header";
import { render, shallow } from "enzyme";
import { MemoryRouter } from "react-router";

describe("Header component", () => {
  it("should render snapshot with default values", () => {
    const wrapper = render(
      <MemoryRouter initialEntries={["/random"]}>
        <WrappedHeader />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should change the menu state on button click", () => {
    const historyMock = { push: jest.fn() };
    const wrapper = shallow(<Header isLogedIn={false} history={historyMock} />);

    wrapper.find(".navbar-toggle").simulate("click");
    expect(wrapper.state()).toHaveProperty("showMenu", true);
  });
});
