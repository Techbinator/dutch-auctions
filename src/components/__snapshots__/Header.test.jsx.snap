// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render snapshot with default values 1`] = `
<header
  class="header"
>
  <div
    class="logo"
  >
    dutch
    <span>
      .auction
    </span>
  </div>
  <div
    class="navbar-toggle"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-bars fa-w-14 "
      data-icon="bars"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
        fill="currentColor"
      />
    </svg>
  </div>
  <nav
    class=""
  >
    <a
      href="/login"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-sign-in-alt fa-w-16 "
        data-icon="sign-in-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
          fill="currentColor"
        />
      </svg>
      Log In
    </a>
    <a
      href="/register"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-registered fa-w-16 "
        data-icon="registered"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.363 207.475c0 18.6-9.831 28.431-28.431 28.431h-29.876v-56.14h23.378c28.668 0 34.929 8.773 34.929 27.709zM504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM363.411 360.414c-46.729-84.825-43.299-78.636-44.702-80.98 23.432-15.172 37.945-42.979 37.945-74.486 0-54.244-31.5-89.252-105.498-89.252h-70.667c-13.255 0-24 10.745-24 24V372c0 13.255 10.745 24 24 24h22.567c13.255 0 24-10.745 24-24v-71.663h25.556l44.129 82.937a24.001 24.001 0 0 0 21.188 12.727h24.464c18.261-.001 29.829-19.591 21.018-35.587z"
          fill="currentColor"
        />
      </svg>
      Register
    </a>
  </nav>
</header>
`;
